File: scope_lifetime/loop_scope.cpp
#include <iostream>

void f() {
  for (int i = 0; i<10; i++) {
    std::cout << "i: " << i << "\n";
  }

  // The following statement will cause a compilation error
  // because i is local to the code in the body of the for
  // loop.
  // std::cout << "i: " << i << "\n";
}

int main() {
  f();
}

/*
 * Expected output:
    $ ./a.out
    i: 0
    i: 1
    i: 2
    i: 3
    i: 4
    i: 5
    i: 6
    i: 7
    i: 8
    i: 9
 */

File: scope_lifetime/loop_scope.py
def f():
  for i in range(1, 10):
    print(f"i (in loop body): {i}")
  print(f"i (outside loop body): {i}")

f()

"""
    expected output:
    $ python3 loop_scope.py
    i (in loop body): 1
    i (in loop body): 2
    i (in loop body): 3
    i (in loop body): 4
    i (in loop body): 5
    i (in loop body): 6
    i (in loop body): 7
    i (in loop body): 8
    i (in loop body): 9
    i (outside loop body): 9
"""
